README.rst

```diff
@@ -95,69 +67,127 @@ To build the images and render the chart locally, use `chartpress
     $ pipenv run chartpress
 
 
-Running in stand-alone mode with minikube
+Running in stand-alone mode with ``kind``
 -----------------------------------------
 
 The notebooks service can be run separate from a ``renku`` deployment. In this
-case, it will function simply as an extension of a JupyterHub deployment.
-Since the functionality is contingent on accessing GitLab, JupyterHub must
-first be added as a GitLab OAuth application. The configuration of the
-callbacks should follow:
+case, it will function as a simple API that can launch anonymous user sessions.
+Follow the steps below to do this.
 
-.. code-block::
 
-    <hub-url>/hub/oauth_callback
-    <hub-url>/hub/api/oauth2/authorize
+1. Launch a ``kind`` cluster that maps port 80 and 443 from the host to the ``kind`` cluster.
+This will allow you to access the notebooks api and sessions without having to setup 
+port forwarding with the ``kubectl`` command.
 
-where ``<hub-url>`` should be the full public address of the hub, including the
-``base_url``, if any. Using the provided `minikube-values.yaml` you can use
+.. code-block:: console
 
-.. code-block::
+    cat <<EOF | kind create cluster --name kind --config=-
+    kind: Cluster
+    apiVersion: kind.x-k8s.io/v1alpha4
+    nodes:
+    - role: control-plane
+      kubeadmConfigPatches:
+      - |
+        kind: InitConfiguration
+        nodeRegistration:
+          kubeletExtraArgs:
+            node-labels: "ingress-ready=true"
+      extraPortMappings:
+      - containerPort: 80
+        hostPort: 80
+        protocol: TCP
+      - containerPort: 443
+        hostPort: 443
+        protocol: TCP
+    EOF
+
+2. Install ``ingress-nginx`` and the notebook service helm chart in the kind cluster.
 
-    http://localhost:31212/hub/oauth_callback
-    http://localhost:31212/hub/api/oauth2/authorize
+.. code-block:: console
 
-You can then deploy JupyterHub and the notebooks service with helm:
+    $ VERSION=$(curl https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/stable.txt)
+    $ kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/$VERSION/deploy/static/provider/kind/deploy.yaml
+    $ cd helm-chart
+    $ helm dep update ./renku-notebooks
+    $ helm upgrade --install renku-notebooks ./renku-notebooks \
+      --set "global.anonymousSessions.enabled=true" \
+      --set "gitlab.url=https://renkulab.io/gitlab" \
+      --set "gitlab.registry.host=registry.renkulab.io" \
+      --set "amalthea.scope.namespaces[0]=default" \
```

Comment:
Are you actually sure that this works? You can set an entire array through the helm command line, see https://github.com/helm/helm/issues/1987#issuecomment-280497496

Reply:
Yeah I just went through the instructions again. And I can successfully install the helm chart with that command. Here are the relevant sections of running `helm get values` after installing.

```yaml
USER-SUPPLIED VALUES:
amalthea:
  scope:
    namespaces:
    - default
 ...
ingress:
  annotations:
    kubernetes.io/ingress.class: nginx
  enabled: true
  hosts:
  - localhost
```